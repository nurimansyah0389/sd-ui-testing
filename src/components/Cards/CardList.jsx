import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import './style.css';

const CardList = ({ data, children }) => <div className="Card">{data.map((item, index) => <Fragment key={index}>{children(item)}</Fragment>)}</div>;
CardList.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string,
        flightName: PropTypes.string,
        departureCode: PropTypes.string,
        departureTime: PropTypes.number,
        arrivalCode: PropTypes.string,
        arrivalTime: PropTypes.number,
        tripDuration: PropTypes.number,
        price: PropTypes.number
    })),
    children: PropTypes.func
};
export default CardList;