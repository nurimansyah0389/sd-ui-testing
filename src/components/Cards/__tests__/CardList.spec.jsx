import React from 'react';
import Renderer from 'react-test-renderer';

import CardList from '../CardList';

describe('Card List Spec', () => {
    it('Should render children without error', () => {
        const tree = Renderer.create(<CardList data={[]}>{() => <div>CHILDREN</div>}</CardList>);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});