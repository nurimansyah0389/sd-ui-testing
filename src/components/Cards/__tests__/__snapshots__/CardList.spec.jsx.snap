// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List Spec Should render children without error 1`] = `
<div
  className="Card"
/>
`;
